src/sm_bluesky/common/electron_analyser/energy_source.py

```diff
@@ -0,0 +1,23 @@
+from dodal.devices.electron_analyser.abstract import (
+    AbstractBaseRegion,
+)
+from ophyd_async.core import DeviceVector
+from ophyd_async.epics.motor import Motor
+
+# This is needed as a workaround until this is fixed
+# https://github.com/bluesky/ophyd-async/issues/903
+key_to_index = {"source1": 0, "source2": 1}
+
```

Comment:
nit: I think this is fine, but every time I does this, I was ask to use enum:
```suggestion
class Source(int, Enum):
    SOURCE1 = 0
    SOURCE2 = 1
```
So that we can forgo the try and catch 
```python
def get_energy_source_for_region(
    region: AbstractBaseRegion, energy_sources: DeviceVector[Motor]
) -> Motor:
    source_alias: str = region.excitation_energy_source
    return energy_sources[Source[source_alias].value]
```
But there is a catch, you have to case match or you will have to override the method for _missing_ to make it case insensitive.